chainservice/chainservice.go

```diff
@@ -163,6 +166,27 @@ func (cs *ChainService) HandleConsensusMsg(msg *iotextypes.ConsensusMessage) err
 	return cs.consensus.HandleConsensusMsg(msg)
 }
 
+// HandleNodeInfoMsg handles nodeinfo message.
+func (cs *ChainService) HandleNodeInfoMsg(ctx context.Context, peer string, msg *iotextypes.ResponseNodeInfoMessage) error {
+	cs.delegateManager.HandleNodeInfo(ctx, peer, msg)
+	return nil
+}
+
+// HandleRequestNodeInfoMsg handles request node info message
+func (cs *ChainService) HandleRequestNodeInfoMsg(ctx context.Context, peerID string, msg *iotextypes.RequestNodeInfoMessage) error {
+	peers, err := cs.p2pAgent.ConnectedPeers()
```

Comment:
the peer could be acquired from dispatcher, unnecessary to be acquired from p2p layer.